src/http-client.js

```diff
@@ -93,13 +110,36 @@ export class HttpClient {
       transformers[i](this, processor, message);
     }
 
-    promise = processor.process(this, message).then(response => {
-      trackRequestEnd(this, processor);
-      return response;
-    }).catch(response => {
-      trackRequestEnd(this, processor);
-      throw response;
-    });
+    processRequest = (message) => {
+      return processor.process(this, message).then(response => {
+        trackRequestEnd(this, processor);
+        return response;
+      }).catch(response => {
+        trackRequestEnd(this, processor);
+        throw response;
+      });
+    };
+
+    var chain = [ processRequest, undefined ];
+    // Apply interceptors
+    for (let interceptor of this.interceptors) {
```

Comment:
I really like for..of, but I think we're trying to avoid using it because it makes babel generate a bunch of extra code to handle iterators, and we need to try to keep file size low wherever possible.